qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1361,14 +1391,7 @@ def layers(self):
                 return
 
             # Construct a shallow copy of self
-            new_layer = DAGCircuit()
-            new_layer.name = self.name
-
-            # add in the registers - this adds the input/output nodes
-            for creg in self.cregs.values():
-                new_layer.add_creg(creg)
-            for qreg in self.qregs.values():
-                new_layer.add_qreg(qreg)
+            new_layer = self._copy_circuit_metadata()
```

Comment:
Good catch here, we probably should have updated these when the `_copy_circuit_metadata` method was added.